packages/@webex/plugin-meetings/src/statsAnalyzer/index.ts

```diff
@@ -493,261 +489,303 @@ export class StatsAnalyzer extends EventsScope {
-              currentStats.framesEncoded === 0
-            ) {
-              LoggerProxy.logger.info(
-                `StatsAnalyzer:index#compareLastStatsResult --> No ${mediaType} frames getting encoded`
-              );
-            }
-
-            if (
-              this.statsResults.resolutions[mediaType].send.framesSent ===
-                this.lastStatsResults.resolutions[mediaType].send.framesSent ||
-              this.statsResults.resolutions[mediaType].send.framesSent === 0
-            ) {
-              LoggerProxy.logger.info(
-                `StatsAnalyzer:index#compareLastStatsResult --> No ${mediaType} frames sent`
-              );
-            }
           }
         }
+      }
 
-        if (this.meetingMediaStatus.expected.sendShare && mediaType.includes('video-share-recv')) {
-          // TODO:need to check receive share value
-          // compare share stats received
-          currentStats = this.statsResults[mediaType].recv;
-          previousStats = this.lastStatsResults[mediaType].recv;
+      if (this.meetingMediaStatus.expected.sendShare) {
+        // TODO:need to check receive share value
+        // compare share stats received
+        const currentPacketsReceived = getCurrentStatsTotals(
+          'video-share-recv',
+          'totalPacketsReceived'
+        );
+        const currentPacketsSent = getCurrentStatsTotals('video-share-recv', 'totalPacketsSent');
+        const previousPacketsReceived = getPreviousStatsTotals(
+          'video-share-recv',
+          'totalPacketsReceived'
+        );
+        const currentFramesReceived = getCurrentResolutionsStatsTotals(
+          'video-share-recv',
+          'totalFramesReceived'
+        );
+        const previousFramesReceived = getPreviousResolutionsStatsTotals(
+          'video-share-recv',
+          'totalFramesReceived'
+        );
+        const currentFramesDecoded = getCurrentStatsTotals(
+          'video-share-recv',
+          'totalFramesDecoded'
+        );
+        const previousFramesDecoded = getPreviousStatsTotals(
+          'video-share-recv',
+          'totalFramesDecoded'
+        );
+        const currentFramesDropped = getCurrentResolutionsStatsTotals(
+          'video-share-recv',
+          'totalFramesDropped'
+        );
+        const previousFramesDropped = getPreviousResolutionsStatsTotals(
+          'video-share-recv',
+          'totalFramesDropped'
+        );
 
-          if (
-            currentStats.totalPacketsReceived === previousStats.totalPacketsReceived ||
-            currentStats.totalPacketsSent === 0
-          ) {
+        const senderFramesDecoded = Object.keys(this.statsResults)
+          .filter((keys) => keys.includes('video-share-recv'))
+          .reduce(
+            (prev, cur) =>
+              prev +
+              ((this.statsResults.resolutions[cur].send &&
+                this.statsResults.resolutions[cur].send.framesDecoded) ||
```

Comment:
`send.framesDecoded` - this doesn't seem right, "send" and "frames decoded"? if we're sending then we're encoding not decoding frames, and the filter in line 751 is for "video-share-recv" while this whole block is for setting a value of `senderFramesDecoded` which is confusing me further

Reply:
Looking at it again I realized the old version was a mistake and the check should have been `this.statsResults.resolutions[mediaType].recv.framesDecoded === 0`.

Reply:
overall there has been quite a few various mismatches in this compareLastStatsResult() function found in this PR, so looks like we don't have any tests for this. I guess, it's because most of this stuff just results in some logs, but maybe it would be worth adding some tests or at least use the sample app and check if the logs this function produces are correct and we don't log "no frame received" or something like that when video is clearly received.

Reply:
The current tests do test `compareLastStatsResult` but only to trigger the start and stopped events.  I tested with the sample app to make sure the stats were coming in correctly, led to my last updates.